Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -842,22 +842,45 @@ protected override void ProcessRecord()
     ///   <para>Update the contents of the Storage Object with the string "OK".</para>
     ///   <para><code>PS C:\> "OK" | Write-GcsObject -Bucket "widget-co-logs" -ObjectName "status.txt"</code></para>
     /// </example>
+    /// <example>
+    ///   <para>Update the contents of the Storage Object piped from Get-GcsObject.</para>
+    ///   <para><code>
+    ///   PS C:\> Get-GcsObject -Bucket "widget-co-logs" -ObjectName "status.txt" | Write-GcsObject -Value "OK"
+    ///   </code></para>
+    /// </example>
     /// </summary>
     [Cmdlet(VerbsCommunications.Write, "GcsObject"), OutputType(typeof(Object))]
     public class WriteGcsObjectCmdlet : GcsObjectCmdlet
     {
         private class ParameterSetNames
         {
-            public const string FromString = "FromString";
-            public const string FromFile = "FromFile";
+            // Write the content of a string to a GCS Object supplied directory to the cmdlet.
+            public const string ByObjectFromString = "ByObjectFromString";
+            // Write the content of a file to a GCS Object supplied directory to the cmdlet.
+            public const string ByObjectFromFile = "ByObjectFromFile";
+            // Write the content of a string to a GCS Object found using Bucket and ObjectName parameter.
+            public const string ByNameFromString = "ByNameFromString";
+            // Write the content of a file to a GCS Object found using Bucket and ObjectName parameter.
+            public const string ByNameFromFile = "ByNameFromFile";
         }
 
         /// <summary>
         /// <para type="description">
+        /// The Google Cloud Storage object to write to.
+        /// </para>
+        /// </summary>
+        [Parameter(ParameterSetName = ParameterSetNames.ByObjectFromString, Mandatory = true, ValueFromPipeline = true)]
```

Comment:
For the new parameter sets, could we make this the first positional parameter? Or would that not make sense?